src/main/java/uk/gov/ida/verifyserviceprovider/factories/VerifyServiceProviderFactory.java

```diff
@@ -117,35 +118,49 @@ private TranslateSamlResponseResource getTranslateMatchingSamlResponseResource()
     }
 
     private TranslateSamlResponseResource getTranslateNonMatchingSamlResponseResource() {
-        VerifyAssertionTranslator verifyAssertionService = responseFactory.createVerifyIdpAssertionService(
-                getHubSignatureTrustEngine(),
-                new SignatureValidatorFactory(),
-                dateTimeComparator,
-                configuration.getHashingEntityId()
-        );
+        AssertionTranslator assertionTranslator = isEidasEnabled()
+            ? createVerifyAndEidasClassifyingAssertionTranslator()
+            : createVerifyAssertionTranslator();
 
-        AssertionTranslator nonMatchingAssertionTranslator;
-        if(isEidasEnabled()) {
-            EidasAssertionTranslator eidasAssertionService = responseFactory.createEidasAssertionService(
-                dateTimeComparator,
-                getEidasMetadataResolverRepository(),
-                configuration.getEuropeanIdentity().get(),
-                configuration.getHashingEntityId()
-            );
-            nonMatchingAssertionTranslator = new ClassifyingAssertionTranslator(verifyAssertionService, eidasAssertionService);
-        } else {
-            nonMatchingAssertionTranslator = verifyAssertionService;
-        }
+        ExplicitKeySignatureTrustEngine signatureTrustEngine = isEidasEnabled()
+            ? createVerifyAndEidasClassifyingSignatureTrustEngine()
+            : getHubSignatureTrustEngine();
 
         return new TranslateSamlResponseResource(
             responseFactory.createNonMatchingResponseService(
-                getHubSignatureTrustEngine(),
-                nonMatchingAssertionTranslator,
+                signatureTrustEngine,
+                assertionTranslator,
                 dateTimeComparator
             ),
             entityIdService);
     }
 
+    private ExplicitKeySignatureTrustEngine createVerifyAndEidasClassifyingSignatureTrustEngine() {
```

Comment:
Why not just `createVerifyAndEidasSignatureTrustEngine`? What does the `Classifying` bit add?

Reply:
Done 👍